src/Command/MigrateCommand.php

```diff
@@ -231,10 +318,66 @@ static function (SplFileInfo $current, $key, $iterator) use ($exclude) {
 
     /**
      * @param string $path
-     * @param string[] $excludePaths
      * @return callable
      */
-    private function createExclusionChecker($path, array $excludePaths)
+    private function createExclusionChecker($path, InputInterface $input)
+    {
+        $exclusions = [];
+        $exclusions = $this->createFileFilter($input->getOption('filter'), $path, $exclusions);
+        $exclusions = $this->createDirectoryExclusionChecker($input->getOption('exclude'), $path, $exclusions);
+        $exclusions = $this->createFileExclusionChecker($input->getOption('exclude-file'), $path, $exclusions);
+
+        return static function ($path) use ($exclusions) {
+            foreach ($exclusions as $exclude) {
+                if ($exclude($path)) {
+                    return true;
+                }
+            }
+            return false;
+        };
+    }
+
+    /**
+     * @param string[] $excludePaths
+     * @param string $path
+     * @param callable[] $exclusions
+     * @return array The $exclusions array with the new exclusion filter
+     *     appended, if created.
+     */
+    private function createFileFilter(array $filters, $path, array $exclusions)
+    {
+        if (empty($filters)) {
+            return $exclusions;
+        }
+
+        /**
+         * @param string $path
+         * @return bool
+         */
+        $exclusions[] = static function ($path) use ($filters) {
+            foreach ($filters as $filter) {
+                $pattern = sprintf('#%s#', $filter);
```

Comment:
I believe we should use `preg_quote` (with delimiter, 2nd arg)

Reply:
I initially did, but this led to all sorts of issues. Dots, slashes, parens, pipes, brackets, and braces were all quoted, which meant that any attempts at quantifiying, using character classes, or grouping would fail.

Reply:
Figured out a solution!